internal/proto/message.go

```diff
@@ -70,6 +70,13 @@ func NewClientMessage(buffer []byte, payloadSize int) *ClientMessage {
 	return clientMessage
 }
 
+func (m *ClientMessage) CloneMessage() *ClientMessage {
+	copiedBuffer := append([]byte{}, m.Buffer...)
```

Comment:
Why do you used `append` here?

Reply:
It turns out that I don't need to use generics. I have updated the prd to use copy instead.